src/Form/OgMenuInstanceForm.php

```diff
@@ -348,10 +388,10 @@ public function save(array $form, FormStateInterface $form_state) {
     }
     $field_storage = FieldStorageConfig::loadByName($this->entity->getEntityTypeId(), OgGroupAudienceHelper::DEFAULT_FIELD);
     $target_type = $field_storage->getSetting('target_type');
-    $group_entity = $this->entity->get(OgGroupAudienceHelper::DEFAULT_FIELD)->getValue();
+    $group_entity = $this->entity->getGroup();
     // If possible, redirect to the group after save.
     if ($target_type && $group_entity) {
-      $form_state->setRedirect('entity.' . $target_type . '.canonical', [$target_type => $group_entity[0]['target_id']]);
+      $form_state->setRedirect('entity.' . $target_type . '.canonical', [$target_type => $group_entity]);
```

Comment:
The full object is passed in here but I think it should be the id only, e.g.
$form_state->setRedirect('entity.' . $target_type . '.canonical', [$target_type => $group_entity->id()]);